module/documents/item.mjs

```diff
@@ -903,6 +904,46 @@ export default class Item5e extends Item {
     return cardData;
   }
 
+  /**
+   * Prepare an object of possible values for item usage.
+   * @returns {ItemUseConfiguration}  Configuration data for the roll.
+   */
+  _getUsageConfig() {
+    const consume = this.system.consume || {};
+    const active = !!this.system.activation?.type;
+
+    const config = {
+      consumeSlot: this.usageScaling === "slot",
+      consumeUses: active && this.hasLimitedUses,
+      consumeResource: active && !!consume.type && !!consume.target && (!this.hasAttack || (consume.type !== "ammo")),
+      createTemplate: active && game.user.can("TEMPLATE_CREATE") && this.hasAreaTarget
+    };
+
+    // Do not suggest consuming your own uses if also consuming them through resources.
+    if ( config.consumeResource && (consume.target === this.id) ) config.consumeUses = false;
+
+    return config;
+  }
```

Comment:
It might not be feasible for various reasons that I hadn't considered, but, as discussed on discord, can we consider a design where we do not need both `_getUsageConfig` and `_getUsageConfigValues` but instead roll them into one? Either the resulting config simply omits an option if it does not apply to the item, or sets it to some special value (like `null`).

I would think then the logic for the `AbilityUseDialog` becomes quite simple: If there are no non-`null` options in the config (or no options at all), the dialog can be skipped. If a value is `true`, it is present but checked by default, and if it is `false`, it is present but unchecked by default.

Reply:
This seems to work.